hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -200,6 +200,20 @@
       milestone: V2
       tests: ["flash_ctrl_error_prog_type"]
     }
+    {
+      name: read_write_overflow
+      desc: '''
+            Send following error transactions with normal traffic and see
+            any catastrophic event happens.
+              - Program flash size longer than 64 bytes
+              - Read flash from controller without settting start op
+              - Issue rd_fifo read more thatn CONTROL.NUM field value.
+            Each transaction is terminated gracefully and should not cause
+            data path lock up.
```

Comment:
Do you also plan to check error code or status?

Reply:
Let me think about this. Original purpose was to focus on survival traffic.
It is depends on whether the error transaction can contaminate normal traffic.
If that is the case, there are so many unpredictable error and we can see error status 
all the time. If dut can isolate the error traffic itself, then I can predict error status
for the error transaction and add check.
I will get back to this thread.

Reply:
so if you try to program larger than the program resolution, you should get a transaction error (i think it's called the prog resolution error).

If you try to read the read fifo before issuing a read, or more than the number of issued reads, you should get an in-band bus error.

Let me know if any of this is not true.

Reply:
o i think i might have misunderstand your program error. Are you saying you send more than what's set in CTRL.NUM?

Reply:
In fact, I send more than 64 bytes. Let me update description for oversized write.
and it gives porg_win error.
Let me update here about more error behavior.

Problem for me, was, this can disturb normal traffic if I mixed with normal
read and write. There is no lockup but more payload mismatch than
error I asserted. I will give you more detail in this thread after I dig further.

Reply:
hmm.... the program should clear the fifo on transaction done. So if this impacts transactions afterwards that would probably be a bug. 

The read side i think should just create errors.

Reply:
@tjaychen ,  @cindychip 
Both errors don't affect subsequent transactions.
oversize write : trigger prog_win_err,
oversize read : set d_err at tl response with all F's.

Reply:
Test is updated to check error for each transaction.

Reply:
Awesome Jaedon, @jdonjdon do you mind also updating this testplan then?

Reply:
sure, updated.